common/src/main/java/com/sequenceiq/cloudbreak/dns/EnvironmentBasedDomainNameProvider.java

```diff
@@ -0,0 +1,63 @@
+package com.sequenceiq.cloudbreak.dns;
+
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class EnvironmentBasedDomainNameProvider {
+    static final String FAILED_TO_GENERATE_DOMAIN_MSG = "Failed to generate domain: The environment and/or account name contains invalid characters!";
+
+    static final String NAMES_SHOULD_BE_SPECIFIED_MSG = "Environment and account names should be specified!";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EnvironmentBasedDomainNameProvider.class);
+
+    private static final String DOMAIN_PART_DELIMITER = ".";
+
+    private static final String DOMAIN_PATTERN = "^((?=[a-z0-9-]{1,63}.)(xn--)?[a-z0-9]+(-[a-z0-9]+)*.)+[a-z]{2,63}$";
```

Comment:
I suppose `.` should be escaped as it was probably meant literally (in both occasions), not as an "any character" wildcard. The part `(xn--)?` is not clear to me, but I can live with it. 😄

Reply:
Thanks for catching it! :D
[It is based on an OReilly pattern for internationalized domain names](https://www.oreilly.com/library/view/regular-expressions-cookbook/9781449327453/ch08s15.html).

Reply:
Ah, that convention is totally new to me. 😮 Then it makes perfectly sense! 👍